README.md

```diff
@@ -154,20 +153,98 @@ for the assessment of each benchmark test:
+after processing the data with GNU R and the script
+[plots.R](scripts/plots.R).
+
+[Figure 2](analysis/files-fig2.pdf) is a copy of the figure submitted
+for publication.  The read and write performance of pixeldata and
+metadata are compared for three software implementations and three
+datasets.  The relative performance of Linux implementations are
+normalized based upon the Java/Linux results, while the Windows
+implementations are normalized based upon the Java/Windows results.
+The read and write performance of C++ is faster than Java in all
+cases.  However, while the JNI read performance is equivalent to Java,
+the JNI write performance is much slower, indicating performance
+problems which scaled with dataset size, approaching two orders of
+magnitude slower with the largest dataset.  C++ metadata read
+performance was faster than Java for datasets with minimal metadata,
+but slower with datasets containing large amounts of metadata.  C++
+metadata write performance was consistently faster than Java.  JNI
+metadata performance was equivalent or very similar to the Java
+performance.  [Supplementary Figure 1](analysis/files-suppfig1.pdf)
+presents the same data with the Implementation and Test categories
+swapped, to allow for easier comparison of implementation differences.
+[Supplementary Figure 2](analysis/files-suppfig2.pdf) shows the
+absolute timings for the above two figures, including the Java/Linux
+and Java/Windows results previously omitted due to normalisation.
+
+Summary data is presented in these tables:
+
+- [pixeldata](analysis/summary-pixeldata-separate.tsv)
+- [metadata](analysis/summary-metadata-separate.tsv)
+
+Additionally, we have reproduced the benchmark by repeating the same
+number of iterations of each test in a loop within the same
+environment. We include these results for completeness and to indicate
+the performance achieved if the same operation is invoked within a
+process multiple times.  [Supplementary Figure
+3](analysis/files-suppfig3.pdf) and [Supplementary Figure
+4](analysis/files-suppfig4.pdf) show the test results with repeated
+iterations within the same environment.  These may be compared with
+Supplementary Figures 1 and 2.  The pixeldata results are almost
+identical with no significant changes.  The metadata results however
+show an improvement in Java performance, with C++ metadata reading
+being slower than Java in all cases, while C++ retained a slight edge
+over Java when writing.  JNI performance was faster than Java for the
+simplest dataset, and marginally faster for datasets with more complex
+metadata.
+
+For most tests, the results were found to be identical and independent
+of whether the tests were run separately or repeated. Interestingly,
+in the case of the metadata tests using Bio-Formats and the Java
+Virtual Machine (JVM), there is a gain due to the optimizer in the
+JVM.  This gain is small for pixeldata, but large for metadata.
```

Comment:
With the extended discussion on the results, this paragraph feels a bit out of place. I could see it either integrated in the previous paragraph (when discussing the Java observations) and/or as an additional bullet point in the conclusions.